### ISSUE TITLE
[Explore] Display log options in a collapsible component

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#82863]
### Discussed in https://github.com/grafana/grafana/discussions/63005

<div type='discussions-op-text'>

<sup>Originally posted by **svennergr** February  7, 2023</sup>
Log options in Explore are currently displayed like this:
![image](https://user-images.githubusercontent.com/8092184/217218992-2dd57609-6b10-43cb-877b-99ba4d22f2a0.png)

By this it's hard to get descriptions about the meaning of those options, e.g. what is "Dedup" exactly doing. Additionally the existing options already take up a good amount of horizontal space.
The idea here would be to create a collapsible component, which holds the options. 
All options are visible and have some kind of description (e.g. via tooltips) when the panel is expaned, only some options/states are visible when the panel is collapsed.

Related issues:
- https://github.com/grafana/grafana/pull/62941#issuecomment-1419427488</div>

**Comment by Elfo404** [comment]
@grafana/observability-logs if i remember correctly you recently mentioned you wanted to revamp this? i'll remove it from our board, but feel free to ping us if you need anything from us

**Comment by matyax** [comment]
Yes, we have a new design concept for Logs in Explore, which should improve how we display those options.
